libp2p/builders.nim

```diff
@@ -187,6 +192,9 @@ proc build*(b: SwitchBuilder): Switch
   if isNil(b.rng):
     b.rng = newRng()
 
+  if isNil(b.peerStore):
```

Comment:
If `PeerStore` is required now, we don't need a `withPeerStore` anymore, at least not one that takes an instance only the non-default params needed to construct it? Otherwise, this construction flow doesn't make much sense.